searchcore/src/vespa/searchcore/proton/server/proton.cpp

```diff
@@ -725,8 +727,16 @@ Proton::ping(std::unique_ptr<MonitorRequest>, MonitorClient &)
     BootstrapConfig::SP configSnapshot = getActiveConfigSnapshot();
     const ProtonConfig &protonConfig = configSnapshot->getProtonConfig();
     ret.distribution_key = protonConfig.distributionkey;
-    ret.timestamp = (_matchEngine->isOnline()) ? 42 : 0;
-    ret.activeDocs = (_matchEngine->isOnline()) ? getNumActiveDocs() : 0;
+    if (_matchEngine->isOnline()) {
+        ret.timestamp = 42;
+        auto [active, targetActive] = getNumActiveDocs();
+        ret.activeDocs = active;
+        ret.targetActiveDocs = targetActive;
+    } else {
+        ret.timestamp = 0;
+        ret.activeDocs = 0;
+        ret.targetActiveDocs = 0; // TODO vekterli hmm... or target anyway ...
```

Comment:
At least for the case where we have redundancy > 1 per group, the change in sum of `targetActiveDocs` reported from _other_ nodes should quickly converge to that of the offline node's total active doc set, so to my ears it sounds the most correct to have this be zero; otherwise we'd risk double-counting the delta between active and target in the group as a whole (well, depending on how these numbers end up interpreted by the dispatcher). Of course, the big challenge is the effective group redundancy == 1 case...!

It's possible to add a mechanism in which the distributors tell content nodes the set of buckets they will eventually receive in a cluster state, which could be used as a proxy for this number when there are no spare replicas on the node itself. Let's keep the TODO for now while we ponder the matter.

Reply:
Ok